### PR TITLE
changelog for 0.6.0

### PR DESCRIPTION
**Pull request by psychocoderHPC**
This PR is applied to the release branch of 0.6.0 and must be backported after we have released alpaka.

### CODE REVIEW

**Review by sbastrakov — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,8 +4,82 @@ All notable changes to this project will be documented in this file.
...
-## [Unreleased]
+## [0.6.0] - 2021-01-XX
+### Compatibility Changes:
+- support for CUDA 11, 11.1, and 11.2 #1076 #1086 #1147 #1231
+- support for CMake 3.18.0 and 3.19.0 #1087 #1217
+- set minimal HIP version to 3.5 #1110
+- remove CMake HIP module shipped with alpaka #1189
+- set HIP-clang as default compiler for HIP #1113
+- support for NVCC + VS 2019 #1121
+- remove support for CUDA 11.0 with MSVC 2019 #1227
```
Comment (bernhardmgruber) [inline CHANGELOG.md]: That should be together with the first line stating that we added support for CUDA 11. E.g.:
`support for CUDA 11 (except MSVC 2019), 11.1, and 11.2 #1076 #1086 #1147 #1231 #1227`

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,8 +4,82 @@ All notable changes to this project will be documented in this file.
...
+- support for boost-1.74.0 #1142
+- explicitly require backends and do not enable them by default #1111
+- remove support for Xcode 11.1 #1206 
+- support Xcode 11.21 - 12.2.0 #1206
+- update to Catch 2.13.3 #1215
+
+### Bug Fixes:
+- apply some clang-tidy fixes #1044
+- fix CUDA/HIP accelerator concept usage #1064
+- fix INTEL compiler detection #1070
```
Comment (bernhardmgruber) [inline CHANGELOG.md]: I would spell "Intel".

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,8 +4,82 @@ All notable changes to this project will be documented in this file.
...
+- CMake: build type CXX flag not passed to nvcc #1073
+- work around Intel ICE when using std::decay on empty template parameter packs #1074
```
Comment (bernhardmgruber) [inline CHANGELOG.md]: What is Intel ICE?

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,8 +4,82 @@ All notable changes to this project will be documented in this file.
...
+- BoostPredef.hpp: Add redefinition of BOOST_COMP_PGI #1082
+- fix min/max return type deduction #1085
+- CMake: fix boost fiber linking #1088
+- fix HIP-clang compile #1107
+- fix CUDA/HIP cmake flags #1152
+- fix error handling CUDA/HIP #1108
+- ALPAKA_DECAY_T: Fix INTEL detection, Add PGI #1116
```
Comment (bernhardmgruber) [inline CHANGELOG.md]: Intel

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,8 +4,82 @@ All notable changes to this project will be documented in this file.
...
+- use std::invoke_result_t instead of std::result_of_t when available #1047
+- simplify shared memory usage in tests #1075 
+- remove boost::aligned_alloc #1094
+- add unit tests for work div #1095
+- change examples (except reduce) to use getValidWorkDiv #1104
+- example monte-carlo-integration #1106 
+- invoke docker run only once instead of twice #1109
+- cpu/SysInfo.hpp: Add #else for cpuid; Add PGI #1119
+- Pgi std atomic workaround #1120
+- BlockSharedMemDynMember::staticAllocBytes: -> function #1118
```
Comment (bernhardmgruber) [inline CHANGELOG.md]: I would rephrase this as `make BlockSharedMemDynMember::staticAllocBytes a function`.

File: CHANGELOG.md (hunk 3 above)
Comment (BenjaminW3) [reply CHANGELOG.md]: ICE is the official compiler speak for Internal Compiler Error. The abbreviation is used in the error messages you get from the compilers.

Comment (bernhardmgruber) [reply CHANGELOG.md]: Now that you explained it, I think I have heard that before! Thx!

Comment (sbastrakov) [reply CHANGELOG.md]: Yes, either that or a German train. But maybe for change log we should actually put a full name for clarity

**Comment by psychocoderHPC** [comment]
I applied the changes and set the release date to 20.01.2021.

**Comment by psychocoderHPC** [comment]
@sbastrakov could you please merge this PR
